### PR TITLE
add h3 spatial index

### PR DESCRIPTION
**Pull request by MarcoGorelli**

### CODE REVIEW

**Comment by ion-elgreco** [comment]
@MarcoGorelli your PR looks ready, any reason it is still in draft?

**Comment by MarcoGorelli** [comment]
I probably just forgot 😄

File: polars_hash/polars_hash/__init__.py [hunk 1]
```diff
@@ -178,6 +178,21 @@ def neighbors(self) -> pl.Expr:
         )
 
 
+@pl.api.register_expr_namespace("h3")
+class H3NameSpace:
+    def __init__(self, expr: pl.Expr):
+        self._expr = expr
+
+    def from_coords(self, len: int = 12) -> pl.Expr:
+        """Takes Struct with latitude, longitude as input and returns utf8 hash using geohash."""
```
Comment (MarcoGorelli) [inline polars_hash/polars_hash/__init__.py:187]:
```suggestion
        """Takes Struct with latitude, longitude as input and returns utf8 H3 spatial index."""
```

File: polars_hash/polars_hash/__init__.py [hunk 2]
```diff
@@ -191,6 +206,10 @@ def nchash(self) -> NonCryptographicHashingNameSpace:
     def geohash(self) -> GeoHashingNameSpace:
         return GeoHashingNameSpace(self)
 
+    @property
+    def h3(self) -> GeoHashingNameSpace:
+        return GeoHashingNameSpace(self)
```
Comment (MarcoGorelli) [inline polars_hash/polars_hash/__init__.py:211]:
```suggestion
    def h3(self) -> H3NameSpace:
        return H3NameSpace(self)
```

**Comment by ion-elgreco** [comment]
Thanks @MarcoGorelli 🙏

**Pull request merged**